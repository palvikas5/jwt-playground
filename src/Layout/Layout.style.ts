import { createGlobalStyle } from 'styled-components';
import { device } from '../utils/device';

const GlobalStyle = createGlobalStyle`
  /**
  * http://meyerweb.com/eric/tools/css/reset/
   v2.0 | 20110126
   License: none (public domain)
  */
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    font: inherit;
    font-size: 100%;
    vertical-align: baseline;
  }

  /* HTML5 display-role reset for older browsers */
  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }

  ol,
  ul {
    list-style: none;
  }

  blockquote,
  q {
    quotes: none;
  }

  blockquote::before,
  blockquote::after,
  q::before,
  q::after {
    content: '';
    content: none;
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
  }

  /* ---------------------- CUSTOM STYLES ---------------------- */

  html {
    box-sizing: border-box;
    font-size: 10px;
  }

  body {
    line-height: 1;
    font-size: 1.6rem;
    font-family: 'Nunito', sans-serif;
    background-color: #2f3b52;
  }

  * {
    margin: 0;
    padding: 0;
    box-sizing: inherit;
  }

  .container {
    max-width: 1280px;
    margin: 0 auto;
    padding-left: 10px;
    padding-right: 10px;

    @media ${device.mobileAndAbove} {
      padding-left: 20px;
      padding-right: 20px;
    }

    @media ${device.mediumAndAbove} {
      padding-left: 40px;
      padding-right: 40px;
    }
  }
`;

export { GlobalStyle };
