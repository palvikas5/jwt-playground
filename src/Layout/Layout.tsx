import React from 'react';
import { Header } from '../components/Header/Header';
import { GlobalStyle } from './Layout.style';

interface LayoutProps {
  children: React.ReactNode;
}

const Layout = ({ children }: LayoutProps) => {
  return (
    <>
      <GlobalStyle />
      <Header />
      {children}
    </>
  );
};

export { Layout };
