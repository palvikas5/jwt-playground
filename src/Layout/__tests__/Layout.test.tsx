import React from 'react';
import { render } from '@testing-library/react';
import { Layout } from '../Layout';

describe('layout', () => {
  it('should render successfully', () => {
    expect.assertions(3);

    const { container, getByText } = render(<Layout>Test 1</Layout>);

    expect(getByText('JWT')).toBeInTheDocument();
    expect(getByText('Test 1')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
