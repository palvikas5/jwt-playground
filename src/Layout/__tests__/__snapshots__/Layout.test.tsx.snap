// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should render successfully 1`] = `
.c0 {
  color: white;
}

.c1 {
  padding-top: 15px;
  padding-bottom: 15px;
}

.c2 {
  font-size: 2.5rem;
}

@media (min-width:425px) {

}

@media (min-width:768px) {

}

<div>
  <div
    class="c0"
  >
    <header
      class="c1 container"
    >
      <div
        class="c2"
      >
        JWT
      </div>
    </header>
  </div>
  Test 1
</div>
`;
